src/runner.rs

```diff
@@ -117,322 +139,274 @@ impl<'a, E: Externals> Interpreter<'a, E> {
-			&Opcode::F32Copysign => self.run_copysign::<F32>(context),
-
-			&Opcode::F64Abs => self.run_abs::<F64>(context),
-			&Opcode::F64Neg => self.run_neg::<F64>(context),
-			&Opcode::F64Ceil => self.run_ceil::<F64>(context),
-			&Opcode::F64Floor => self.run_floor::<F64>(context),
-			&Opcode::F64Trunc => self.run_trunc::<F64>(context),
-			&Opcode::F64Nearest => self.run_nearest::<F64>(context),
-			&Opcode::F64Sqrt => self.run_sqrt::<F64>(context),
-			&Opcode::F64Add => self.run_add::<F64>(context),
-			&Opcode::F64Sub => self.run_sub::<F64>(context),
-			&Opcode::F64Mul => self.run_mul::<F64>(context),
-			&Opcode::F64Div => self.run_div::<F64, F64>(context),
-			&Opcode::F64Min => self.run_min::<F64>(context),
-			&Opcode::F64Max => self.run_max::<F64>(context),
-			&Opcode::F64Copysign => self.run_copysign::<F64>(context),
-
-			&Opcode::I32WrapI64 => self.run_wrap::<i64, i32>(context),
-			&Opcode::I32TruncSF32 => self.run_trunc_to_int::<F32, i32, i32>(context),
-			&Opcode::I32TruncUF32 => self.run_trunc_to_int::<F32, u32, i32>(context),
-			&Opcode::I32TruncSF64 => self.run_trunc_to_int::<F64, i32, i32>(context),
-			&Opcode::I32TruncUF64 => self.run_trunc_to_int::<F64, u32, i32>(context),
-			&Opcode::I64ExtendSI32 => self.run_extend::<i32, i64, i64>(context),
-			&Opcode::I64ExtendUI32 => self.run_extend::<u32, u64, i64>(context),
-			&Opcode::I64TruncSF32 => self.run_trunc_to_int::<F32, i64, i64>(context),
-			&Opcode::I64TruncUF32 => self.run_trunc_to_int::<F32, u64, i64>(context),
-			&Opcode::I64TruncSF64 => self.run_trunc_to_int::<F64, i64, i64>(context),
-			&Opcode::I64TruncUF64 => self.run_trunc_to_int::<F64, u64, i64>(context),
-			&Opcode::F32ConvertSI32 => self.run_extend::<i32, F32, F32>(context),
-			&Opcode::F32ConvertUI32 => self.run_extend::<u32, F32, F32>(context),
-			&Opcode::F32ConvertSI64 => self.run_wrap::<i64, F32>(context),
-			&Opcode::F32ConvertUI64 => self.run_wrap::<u64, F32>(context),
-			&Opcode::F32DemoteF64 => self.run_wrap::<F64, F32>(context),
-			&Opcode::F64ConvertSI32 => self.run_extend::<i32, F64, F64>(context),
-			&Opcode::F64ConvertUI32 => self.run_extend::<u32, F64, F64>(context),
-			&Opcode::F64ConvertSI64 => self.run_extend::<i64, F64, F64>(context),
-			&Opcode::F64ConvertUI64 => self.run_extend::<u64, F64, F64>(context),
-			&Opcode::F64PromoteF32 => self.run_extend::<F32, F64, F64>(context),
-
-			&Opcode::I32ReinterpretF32 => self.run_reinterpret::<F32, i32>(context),
-			&Opcode::I64ReinterpretF64 => self.run_reinterpret::<F64, i64>(context),
-			&Opcode::F32ReinterpretI32 => self.run_reinterpret::<i32, F32>(context),
-			&Opcode::F64ReinterpretI64 => self.run_reinterpret::<i64, F64>(context),
+		Ok(RunResult::Return)
+	}
+
+	#[inline(always)]
+	fn run_instruction(&mut self, context: &mut FunctionContext, instruction: &isa::Instruction) -> Result<InstructionOutcome, TrapKind> {
+		match instruction {
+			&isa::Instruction::Unreachable => self.run_unreachable(context),
```

Comment:
Are these `run_*` functions inlined by LLVM? That'd be a really good source of easy wins if not.

Reply:
Not sure about all of them, but I experimented with inlining of hot functions (such as `set_local`/`get_local` for `fac-opt`) and it didn't make a difference.